lib/core/util.js

```diff
@@ -58,31 +58,31 @@ function parseURL (url) {
     throw new InvalidArgumentError('Invalid URL: The URL argument must be a non-null object.')
   }
 
-  if (url.port != null && url.port !== '' && !Number.isFinite(parseInt(url.port))) {
-    throw new InvalidArgumentError('Invalid URL: port must be a valid integer or a string representation of an integer.')
+  if (!/^https?:/.test(url.origin || url.protocol)) {
+    throw new InvalidArgumentError('Invalid URL protocol: the URL must start with `http:` or `https:`.')
   }
 
-  if (url.path != null && typeof url.path !== 'string') {
-    throw new InvalidArgumentError('Invalid URL path: the path must be a string or null/undefined.')
-  }
+  if (!(url instanceof URL)) {
```

Comment:
Small, `nit`.

Does `url.constructor` is slower compared to `instanceof`?

Also, `instance tampering` is possible when using `instanceof`, where using the `well-known` symbol can allow non-valid instances of URL to return `true`.

Just wanted to point that out, not a blocker 🙂

Reply:
Compared to new URL which takes around 300-400ns, I think constructor vs instanceof is negligible